clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/HttpCloudFoundryClient.java

```diff
@@ -104,157 +67,115 @@
-    try {
-      return retry.executeCallable(
-          () -> {
-            Response response = chain.proceed(chain.request());
-            lastResponse.set(response);
-
-            switch (response.code()) {
-              case 401:
-                BufferedSource source = response.body().source();
-                source.request(Long.MAX_VALUE); // request the entire body
-                Buffer buffer = source.buffer();
-                String body = buffer.clone().readString(Charset.forName("UTF-8"));
-                if (!body.contains("Bad credentials")) {
-                  response =
-                      chain.proceed(
-                          chain
-                              .request()
-                              .newBuilder()
-                              .header("Authorization", "bearer " + getToken(true).getAccessToken())
-                              .build());
-                  lastResponse.set(response);
-                }
-                break;
-              case 502:
-              case 503:
-              case 504:
-                // after retries fail, the response body for these status codes will get wrapped up
-                // into a CloudFoundryApiException
-                throw new RetryableApiException(
-                    "Response Code "
-                        + response.code()
-                        + ": "
-                        + chain.request().httpUrl()
-                        + " attempting retry");
-            }
-
-            return response;
-          });
-    } catch (SocketTimeoutException e) {
-      throw new RuntimeException(e);
-    } catch (Exception e) {
-      final Response response = lastResponse.get();
-      if (response == null) {
-        throw new IllegalStateException(e);
-      }
-      return response;
-    }
-  }
-
   public HttpCloudFoundryClient(
       String account,
       String appsManagerUri,
       String metricsUri,
       String apiHost,
       String user,
       String password,
+      boolean useHttps,
       boolean skipSslValidation,
       Integer resultsPerPage,
-      ForkJoinPool forkJoinPool) {
+      ForkJoinPool forkJoinPool,
+      OkHttpClient.Builder okHttpClientBuilder,
+      MeterRegistry registry) {
+
     this.apiHost = apiHost;
     this.user = user;
     this.password = password;
 
-    this.okHttpClient = createHttpClient(skipSslValidation);
-    this.okHttpClient.setReadTimeout(20, TimeUnit.SECONDS);
-
-    okHttpClient.interceptors().add(this::createRetryInterceptor);
-
     ObjectMapper mapper = new ObjectMapper();
     mapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
     mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
     mapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
     mapper.registerModule(new JavaTimeModule());
 
-    this.jacksonConverter = new JacksonConverter(mapper);
+    okHttpClientBuilder.eventListener(
```

Comment:
Let's remove the eventListener for now (also un-autowire the registry/remove it). I found a more "spinnaker" way of doing it and we will be able to leverage OkHttp3MetricsInterceptor from kork-web. However, lets do that in a separate PR.

Reply:
Done